substrate/frame/support/procedural/src/pallet/expand/call.rs

```diff
@@ -345,7 +345,8 @@ pub fn expand_call(def: &mut Def) -> proc_macro2::TokenStream {
 							>::pays_fee(&__pallet_base_weight, ( #( #args_name, )* ));
 
 							#frame_support::dispatch::DispatchInfo {
-								weight: __pallet_weight,
+								call_weight: __pallet_weight,
+								extension_weight: Default::default(),
```

Comment:
This feels a bit error prone. Some people could just call `get_dispatch_info` on the call and expect the same as the `get_dispatch_info` on the extrinsic. And some not careful refactor can simply forget the `extension_weight`.

Maybe we can have 2 struct `ExtrinsicDispatchInfo` and `CallDispatchInfo` (just a rename with deprecated), and if the code is correct it should actually refactor quite fine maybe.

But simpler: we can have a simple struct: with some partial information
```rust
extension_weight: Option<Weight>,
```
and `None` would indicate that no information is present

we make the item `extension_weight` private and expose 1 method:
```rust
fn extension_weight_or_default_log_error(proof: &str) {
if some then return weight
else log error that extension_weight is queried and wrong proof
}
```

Reply:
While I definitely agree with the sentiment, I tried to keep the changes to basic structures like `DispatchInfo` minimal. I would say that a refactor around the dispatch info into 2 pieces would be tougher than expected, because the `Dispatchable` trait provides the `Info` type, which is a call info. The `PostInfo` type would have to similarly be split then into call and extrinsic.

It is possible, but my feeling is that it falls outside the scope of this PR, which is already pretty big on its own. We can do the refactor later if all of the weight is accounted for in this current format.

Reply:
Yes 2 structs sounds too much breaking. but a runtime error logged when `extension_weight` is used when not provided could be good.

Anyway we can do in a follow-up PR

Reply:
I think the issue is also partially in definitions.

What we are talking here is Get_Dispatch_Info. 

We have to first define what is Dispatchable. AFAIK, what is `Dispatchable` is a `Call`, not an `Extrinsic`. So the pedantic long term solution IMO is to remove the `impl GetDispatchInfo for UncheckedExtrinsic`.